### PR TITLE
Offer property attribute target in record positional param

### LINKED ISSUE #51756: Completion don't provide property target in record parameter
**Version Used**: 
VS 16.9.0

**Steps to Reproduce**:

```csharp
public record R([/*here*/] string M);

// modifying attribute target:
public record R([/*here*/ SomeAttribute] string M);

// adding attribute:
public record R([/*here*/ string M);
```

**Expected Behavior**:
Contextual keyword `property` provided.

**Actual Behavior**:
No contextual keyword `property` provided.

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #51756

image

### CODE REVIEW

**Comment by Youssef1313** [comment]
Same failure again 😕

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs [hunk 1]
```diff
@@ -321,11 +321,14 @@ public bool IsMemberAttributeContext(ISet<SyntaxKind> validTypeDeclarations, Can
...
             if (token.Kind() == SyntaxKind.OpenBracketToken &&
-                token.Parent.IsKind(SyntaxKind.AttributeList) &&
-                this.SyntaxTree.IsMemberDeclarationContext(
-                    token.SpanStart, contextOpt: null, validModifiers: null, validTypeDeclarations: validTypeDeclarations, canBePartial: false, cancellationToken: cancellationToken))
+                token.Parent.IsKind(SyntaxKind.AttributeList))
             {
-                return true;
+                var isRecordParameter = token.Parent.IsParentKind(SyntaxKind.Parameter) &&
+                    token.Parent.Parent.IsParentKind(SyntaxKind.ParameterList) &&
+                    token.Parent.Parent.Parent.IsParentKind(SyntaxKind.RecordDeclaration);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs:328]: `if (token.Parent is { Parent: { Kind: Parameter, Parent: { Parent { Kind: RecordDeclaration } } } })`
?

**Review by CyrusNajmabadi — APPROVED** [review]
LMK if you want to make changes.  otherwise i'll merge in.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs:328]: @CyrusNajmabadi I'll have to use RawKind for this form as there is no `Kind` property. The cast to int feels annoying to me. But let me know if this is okay for you.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs [hunk 2]
```diff
@@ -321,11 +321,14 @@ public bool IsMemberAttributeContext(ISet<SyntaxKind> validTypeDeclarations, Can
...
+
+                return isRecordParameter || SyntaxTree.IsMemberDeclarationContext(
+                    token.SpanStart, contextOpt: null, validModifiers: null, validTypeDeclarations: validTypeDeclarations, canBePartial: false, cancellationToken: cancellationToken);
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs:331]: Done

**Comment by Youssef1313** [comment]
@CyrusNajmabadi Anything left here? Thanks.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs [hunk 3]
```diff
@@ -321,11 +321,20 @@ public bool IsMemberAttributeContext(ISet<SyntaxKind> validTypeDeclarations, Can
...
             if (token.Kind() == SyntaxKind.OpenBracketToken &&
-                token.Parent.IsKind(SyntaxKind.AttributeList) &&
-                this.SyntaxTree.IsMemberDeclarationContext(
-                    token.SpanStart, contextOpt: null, validModifiers: null, validTypeDeclarations: validTypeDeclarations, canBePartial: false, cancellationToken: cancellationToken))
+                token.Parent.IsKind(SyntaxKind.AttributeList))
             {
-                return true;
+                if (token.Parent.IsParentKind(SyntaxKind.Parameter) &&
+                    token.Parent.Parent.IsParentKind(SyntaxKind.ParameterList) &&
+                    token.Parent.Parent.Parent.IsParentKind(SyntaxKind.RecordDeclaration))
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs:328]:
```suggestion
                if (token.Parent?.Parent is ParameterSyntax { Parent: ParameterListSyntax { Parent:  RecordDeclaration _ } } )
```

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
lgtm.  thanks!

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs (hunk 3 above)
Comment (Youssef1313) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs:328]:
```suggestion
                if (token.Parent.Parent is ParameterSyntax { Parent: ParameterListSyntax { Parent:  RecordDeclarationSyntax } } )
```

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs [hunk 4]
```diff
@@ -323,9 +323,7 @@ public bool IsMemberAttributeContext(ISet<SyntaxKind> validTypeDeclarations, Can
             if (token.Kind() == SyntaxKind.OpenBracketToken &&
                 token.Parent.IsKind(SyntaxKind.AttributeList))
             {
-                if (token.Parent.IsParentKind(SyntaxKind.Parameter) &&
-                    token.Parent.Parent.IsParentKind(SyntaxKind.ParameterList) &&
-                    token.Parent.Parent.Parent.IsParentKind(SyntaxKind.RecordDeclaration))
+                if (token.Parent.Parent is ParameterSyntax { Parent: ParameterListSyntax { Parent:  RecordDeclarationSyntax } } )
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs:326]:
```suggestion
                if (token.Parent.Parent is ParameterSyntax { Parent: ParameterListSyntax { Parent: RecordDeclarationSyntax } } )
```

**Comment by Youssef1313** [comment]
@CyrusNajmabadi Auto merge was cancelled due to pushing a commit to fix formatting. It should hopefully be green now.

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**